src/main/java/com/epam/lowcost/DAO/FlightDAOImpl.java

```diff
@@ -0,0 +1,153 @@
+package com.epam.lowcost.DAO;
+
+import com.epam.lowcost.model.Flight;
+import com.epam.lowcost.util.DateFormatter;
+import com.epam.lowcost.util.PlaneRowMapper;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FlightDAOImpl implements FlightDAO {
+    private DataSource dataSource;
+
+    public FlightDAOImpl(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+
+    @Override
+    public List<Flight> getAllFlights() {
+        List<Flight> flights = new ArrayList<>();
+        try (Connection conn = dataSource.getConnection();
+             Statement stmt = conn.createStatement();
+             ResultSet rs = stmt.executeQuery("SELECT * FROM FLIGHTS JOIN  PLANES " +
+                     "ON FLIGHTS.plane_id = PLANES.id WHERE  FLIGHTS.ISDELETED = false ")) {
+            while (rs.next()) {
+                flights.add(extractFlightFromRS(rs));
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return flights;
+    }
+
+    @Override
+    public Flight getById(Long id) {
+        Flight flight = null;
+        String sql = String.format("SELECT * FROM FLIGHTS JOIN  PLANES" +
+                " ON FLIGHTS.plane_id = PLANES.id  WHERE FLIGHTS.id = '%d' AND FLIGHTS.isDeleted=FALSE", id);
+        try (Connection conn = dataSource.getConnection();
+             Statement stmt = conn.createStatement();
+             ResultSet rs = stmt.executeQuery(sql)) {
+            if (rs.next()) {
+                return extractFlightFromRS(rs);
+
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return flight;
+    }
+
+    @Override
+    public Flight addNewFlight(Flight flight) {
+        Long price = flight.getInitialPrice();
+        LocalDateTime depatureDate = flight.getDepartureDate();
+        LocalDateTime arrivalDate = flight.getArrivalDate();
+        Long plane_id = flight.getPlane().getId();
```

Comment:
why u use `_` for name ?

Reply:
why not? :)
corrected......

Reply:
https://www.geeksforgeeks.org/java-naming-conventions/
3 item